tests/data_generation/helper/induce_causal_effect.py

```
@@ -1,555 +1,303 @@
-                return subj_df
+        return simulated_df, ite_df
 
-        return self._add_event(subj_df, self.exposure_name, event_time)
+    def _compute_daily_prob(self, total_prob: float, num_days: int) -> float:
+        """Converts a total probability over a period into a daily probability."""
+        if num_days <= 0 or total_prob >= 1.0:
+            return total_prob
+        return 1 - (1 - total_prob) ** (1 / num_days)
 
-    def simulate_outcome(
+    def _simulate_time_to_first_event(
```

Pylint (3.3.7)

[refactor] Too many arguments (7/5)

(R0913)